modules/oe_paragraphs_media/oe_paragraphs_media.post_update.php

```diff
@@ -31,10 +31,31 @@ function oe_paragraphs_media_post_update_00001(array &$sandbox) {
 }
 
 /**
- * Create new form modes for Text with featured Media paragraph.
+ * Rename "Title" field to "Heading".
  */
 function oe_paragraphs_media_post_update_00002(): void {
-  $storage = new FileStorage(drupal_get_path('module', 'oe_paragraphs_media') . '/config/post_updates/00002_create_form_modes');
+  $field_config = FieldConfig::load('paragraph.oe_text_feature_media.field_oe_title');
+  $field_config->setLabel('Heading');
+  $field_config->save();
+}
+
+/**
+ * Add "Title" field to the Text with featured Media paragraph.
+ */
+function oe_paragraphs_media_post_update_00003(): void {
+  // Create "Title" field storage first.
+  module_load_include('php', 'oe_paragraphs', 'oe_paragraphs.post_update');
+  oe_paragraphs_post_update_10007();
```

Comment:
Why do we add the field storage in the main module? If we need a field storage to be used across modules we could create a submodule as we did for `oe_paragraphs_media_field_storage`. I don't think we should add a post_update creating a field storage and then calling the post_update in a submodule...

Reply:
oe_paragraphs stores common field storages in the config/install. You can find there such configurations as field.storage.paragraph.field_oe_text.yml, field.storage.paragraph.field_oe_text_long.yml, field.storage.paragraph.field_oe_title.yml that are used in paragraphs in the component. So field.storage.paragraph.field_oe_text_title.yml  has been placed also there. Not sure that we should create 18 modules for each field storage.

> I don't think we should add a post_update creating a field storage and then calling the post_update in a submodule.
We have to run oe_paragraphs_media_post_update_00003 before oe_paragraphs_post_update_10007. By default it doesn't happen. In current implementation oe_paragraphs_post_update_10007 is invoked twice - first time field storage is created, second time it is updated that is allowed by /core/lib/Drupal/Core/Config/ConfigInstaller.php line 361.
Please suggest another solution if this is inappropriate.

Reply:
Well the field seems strictly needed only for this paragraph and in this case both config files should be part of the module. I'll pass the review to @imanoleguskiza for 2nd opinion.

Reply:
@22Alexandra is right in this situation, the new field is exclusive to the text with featured media so we should keep it contained to the submodule.